src/time_integration/paired_explicit_runge_kutta/methods_PERK2.jl

```diff
@@ -0,0 +1,396 @@
+
+    consistency_order = 2
+
+    dtmax = tspan[2] - tspan[1]
+    dteps = 1e-9 # Hyperparameter of the optimization, might be too large for systems requiring very small timesteps
+
+    num_eig_vals, eig_vals = filter_eig_vals(eig_vals; verbose)
+
+    monomial_coeffs, dt_opt = bisect_stability_polynomial(consistency_order,
+                                                          num_eig_vals, num_stages,
+                                                          dtmax,
+                                                          dteps,
+                                                          eig_vals; verbose)
+    monomial_coeffs = undo_normalization!(monomial_coeffs, consistency_order,
+                                          num_stages)
+
+    num_monomial_coeffs = length(monomial_coeffs)
+    @assert num_monomial_coeffs == coeffs_max
+    A = compute_a_coeffs(num_stages, stage_scaling_factors, monomial_coeffs)
+
+    a_matrix[:, 1] -= A
+    a_matrix[:, 2] = A
+
+    return a_matrix, c
+end
+
+# Compute the Butcher tableau for a paired explicit Runge-Kutta method order 2
+# using provided monomial coefficients file
+function compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                   base_path_monomial_coeffs::AbstractString,
+                                                   bS, c_end)
+
+    # c Vector form Butcher Tableau (defines timestep per stage)
+    c = zeros(num_stages)
+    for k in 2:num_stages
+        c[k] = c_end * (k - 1) / (num_stages - 1)
+    end
+    stage_scaling_factors = bS * reverse(c[2:(end - 1)])
+
+    # - 2 Since first entry of A is always zero (explicit method) and second is given by c_2 (consistency)
+    coeffs_max = num_stages - 2
+
+    a_matrix = zeros(coeffs_max, 2)
+    a_matrix[:, 1] = c[3:end]
+
+    path_monomial_coeffs = joinpath(base_path_monomial_coeffs,
+                                    "gamma_" * string(num_stages) * ".txt")
+
+    @assert isfile(path_monomial_coeffs) "Couldn't find file"
+    monomial_coeffs = readdlm(path_monomial_coeffs, Float64)
+    num_monomial_coeffs = size(monomial_coeffs, 1)
+
+    @assert num_monomial_coeffs == coeffs_max
+    A = compute_a_coeffs(num_stages, stage_scaling_factors, monomial_coeffs)
+
+    a_matrix[:, 1] -= A
+    a_matrix[:, 2] = A
+
+    return a_matrix, c
+end
+
+"""
+    PairedExplicitRK2(num_stages, base_path_monomial_coeffs::AbstractString,
+                      bS = 1.0, c_end = 0.5)
+    PairedExplicitRK2(num_stages, tspan, semi::AbstractSemidiscretization;
+                      verbose = false, bS = 1.0, c_end = 0.5)
+    PairedExplicitRK2(num_stages, tspan, eig_vals::Vector{ComplexF64};
+                      verbose = false, bS = 1.0, c_end = 0.5)
+
+The following structures and methods provide a minimal implementation of
+the second-order paired explicit Runge-Kutta (PERK) method
+optimized for a certain simulation setup (PDE, IC & BC, Riemann Solver, DG Solver).
```

Comment:
Could you please explain more about the arguments of the different constructors in the docstring? E.g. what do `bS` and `c_end` do? This is not clear (to me) from their name. Also `eig_vals` and `base_path_monomial_coeffs` could get some more explanation, I think (e.g. what is the expected format and name of the file in `base_path_monomial_coeffs`).